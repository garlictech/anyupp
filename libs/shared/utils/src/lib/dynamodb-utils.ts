import AWS from 'aws-sdk';

// Creates a dynamodb update expression for Document Client.
// Assumes, that the key name is id...
export const createUpdateParams = (
  TableName: string,
  id: string,
  item: Record<string, unknown>,
): AWS.DynamoDB.UpdateItemInput => ({
  TableName,
  Key: { id: { S: id } },
  UpdateExpression:
    'set ' +
    Object.keys(item)
      .map(k => `#${k} = :${k}`)
      .join(', '),
  ExpressionAttributeNames: Object.entries(item).reduce(
    (acc, cur) => ({ ...acc, [`#${cur[0]}`]: cur[0] }),
    {},
  ),
  ExpressionAttributeValues: Object.entries(item).reduce(
    (acc, cur) => ({ ...acc, [`:${cur[0]}`]: cur[1] }),
    {},
  ),
});
