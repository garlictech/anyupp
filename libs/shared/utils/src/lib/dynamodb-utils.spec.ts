import { createUpdateParams } from './dynamodb-utils';

describe('Test dynamodb utils', () => {
  test('createUpdateParams tests', () => {
    const fixture = {
      prop1: 'prop1',
      prop2: 2,
      prop3: true,
      prop4: {
        embeddedProp: 'embeddedProp',
      },
    };

    expect(
      createUpdateParams('tablename', 'THE_KEY', fixture),
    ).toMatchSnapshot();
  });
});
