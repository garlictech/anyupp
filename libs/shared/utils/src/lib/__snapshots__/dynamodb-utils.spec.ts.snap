// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test dynamodb utils createUpdateParams tests 1`] = `
Object {
  "ExpressionAttributeNames": Object {
    "#prop1": "prop1",
    "#prop2": "prop2",
    "#prop3": "prop3",
    "#prop4": "prop4",
  },
  "ExpressionAttributeValues": Object {
    ":prop1": "prop1",
    ":prop2": 2,
    ":prop3": true,
    ":prop4": Object {
      "embeddedProp": "embeddedProp",
    },
  },
  "Key": Object {
    "id": "THE_KEY",
  },
  "ReturnValues": "ALL_NEW",
  "TableName": "tablename",
  "UpdateExpression": "set #prop1 = :prop1, #prop2 = :prop2, #prop3 = :prop3, #prop4 = :prop4",
}
`;
